weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksEvaluationComparison.ts

```diff
@@ -548,6 +482,532 @@ const fetchEvaluationComparisonResults = async (
+  source: 'derived',
+  scoreType: 'continuous',
+  metricSubPath: ['Total Tokens'],
+  shouldMinimize: true,
+  unit: '',
+};
+const pickColor = (ndx: number) => {
+  return WB_RUN_COLORS[ndx % WB_RUN_COLORS.length];
+};
+
+const isBinaryScore = (score: any): score is boolean => {
+  return typeof score === 'boolean';
+};
+
+const isBinarySummaryScore = (score: any): score is BinarySummaryScore => {
+  return (
+    typeof score === 'object' &&
+    score != null &&
+    'true_count' in score &&
+    'true_fraction' in score
+  );
+};
+
+const isContinuousSummaryScore = (
+  score: any
+): score is ContinuousSummaryScore => {
+  return typeof score === 'object' && score != null && 'mean' in score;
+};
+
+const isContinuousScore = (score: any): score is number => {
+  return typeof score === 'number';
+};
+
+type BinarySummaryScore = {
+  true_count: number;
+  true_fraction: number;
+};
+
+type ContinuousSummaryScore = {
+  mean: number;
+};
+
+type EvaluationEvaluateCallSchema = TraceCallSchema & {
+  inputs: TraceCallSchema['inputs'] & {
+    self: string;
+    model: string;
+  };
+  output: TraceCallSchema['output'] & {
+    [scorer: string]: {
+      [score: string]: SummaryScore;
+    };
+  } & {
+    model_latency: ContinuousSummaryScore;
+  };
+  summary: TraceCallSchema['summary'] & {
+    usage?: {
+      [model: string]: {
+        requests?: number;
+        completion_tokens?: number;
+        prompt_tokens?: number;
+        total_tokens?: number;
+      };
+    };
+  };
+};
+type SummaryScore = BinarySummaryScore | ContinuousSummaryScore;
+
+function fuzzyMatchScorerName(
+  scoreNames: string[],
+  possibleScorerName: string
+) {
+  // anytime we see a '-' in possibleScorerName, it can be any illegal character
+  // in score names. Use a regex to find matches, and return the first match.
+  const regex = new RegExp(possibleScorerName.replace(/-/g, '.'));
+  return scoreNames.find(name => regex.test(name));
+}
+
+// Process summary data specifically for regular evaluations
+const processEvaluationSummary = (
+  result: EvaluationComparisonSummary,
+  evalCall: any,
+  evalCallId: string,
+  evalObj: any,
+  output: any
+): void => {
+  // Add the user-defined scores
+  evalObj.scorerRefs.forEach((scorerRef: string) => {
+    const scorerKey = getScoreKeyNameFromScorerRef(scorerRef);
+    // TODO: REMOVE when sanitized scorer names have been released
+    // this is a hack to support previous unsanitized scorer names
+    // that have spaces.
+    let score = output[scorerKey];
+    if (score == null && scorerKey.includes('-')) {
+      // no score found, '-' means we probably sanitized an illegal character
+      const foundScorerNameMaybe = fuzzyMatchScorerName(
+        Object.keys(output),
+        scorerKey
+      );
+      if (foundScorerNameMaybe != null) {
+        score = output[foundScorerNameMaybe];
+      }
+    }
+    const recursiveAddScore = (scoreVal: any, currPath: string[]) => {
```

Comment:
seems like this large block of recursiveAddScore is shared between the implementations?

Reply:
punt for now and payback later (or move to the server)